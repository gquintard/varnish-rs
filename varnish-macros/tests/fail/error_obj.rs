struct Obj;
struct ObjVclNameTy;
struct ObjVclNameDup;
struct ObjGen<T> {
    _marker: std::marker::PhantomData<T>,
}

#[varnish::vmod]
mod err {
    use super::*;

    impl super::Obj {}
    impl<T> ObjGen<T> {}

    impl Obj {
        fn non_public() {}

        pub async fn async_fn() {}

        #[event]
        pub fn event_fn() {}
    }

    impl ObjVclNameTy {
        pub fn new(#[vcl_name] a: String) {}
    }

    impl ObjVclNameDup {
        pub fn new(#[vcl_name] a: &str, #[vcl_name] b: &str) {}
    }
}

fn main() {}
