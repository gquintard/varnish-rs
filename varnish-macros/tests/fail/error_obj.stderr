error: Expected a simple type for object. If the object is defined elsewhere, use `use` to import it.
  --> tests/fail/error_obj.rs:12:10
   |
12 |     impl super::Obj {}
   |          ^^^^^

error: Object must have a constructor called `new`
  --> tests/fail/error_obj.rs:12:10
   |
12 |     impl super::Obj {}
   |          ^^^^^

error: Generics are not supported for object impls
  --> tests/fail/error_obj.rs:13:10
   |
13 |     impl<T> ObjGen<T> {}
   |          ^

error: Object must have a constructor called `new`
  --> tests/fail/error_obj.rs:13:13
   |
13 |     impl<T> ObjGen<T> {}
   |             ^^^^^^

error: Only public functions and impl blocks are allowed inside a `mod` tagged with `#[varnish::vmod]`. Add `pub` or move this function outside of this mod.
  --> tests/fail/error_obj.rs:16:9
   |
16 |         fn non_public() {}
   |         ^^

error: async functions are not supported
  --> tests/fail/error_obj.rs:18:13
   |
18 |         pub async fn async_fn() {}
   |             ^^^^^

error: Event functions are not supported for object methods
  --> tests/fail/error_obj.rs:20:11
   |
20 |         #[event]
   |           ^^^^^

error: Object must have a constructor called `new`
  --> tests/fail/error_obj.rs:15:10
   |
15 |     impl Obj {
   |          ^^^

error: #[vcl_name] params must be declared as `&str`
  --> tests/fail/error_obj.rs:25:32
   |
25 |         pub fn new(#[vcl_name] a: String) {}
   |                                ^

error: Object must have a constructor called `new`
  --> tests/fail/error_obj.rs:24:10
   |
24 |     impl ObjVclNameTy {
   |          ^^^^^^^^^^^^

error: #[vcl_name] param is allowed only once in a function args list
  --> tests/fail/error_obj.rs:29:53
   |
29 |         pub fn new(#[vcl_name] a: &str, #[vcl_name] b: &str) {}
   |                                                     ^

error: Object must have a constructor called `new`
  --> tests/fail/error_obj.rs:28:10
   |
28 |     impl ObjVclNameDup {
   |          ^^^^^^^^^^^^^

error: No functions or objects found in this module
 --> tests/fail/error_obj.rs:9:5
  |
9 | mod err {
  |     ^^^
